url20b0f880a5_jgraph_legacy_jgraph5_tgz-pJ8-org_jgraph_example_HelloWorldJ8/src/com/jgraph/layout/svg/SVGExample.java

```diff
@@ -0,0 +1,156 @@
+ * installation directory. If you are unable to locate this file please
+ * contact JGraph sales for another copy.
+ */
+package com.jgraph.layout.svg;
+
+import java.awt.Color;
+import java.io.BufferedOutputStream;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.OutputStream;
+import java.net.URL;
+
+import javax.swing.ImageIcon;
+import javax.swing.JFileChooser;
+import javax.swing.JFrame;
+import javax.swing.JOptionPane;
+import javax.swing.filechooser.FileFilter;
+
+import org.jgraph.graph.VertexRenderer;
+import org.jgraph.graph.VertexView;
+
+import com.jgraph.example.GraphEdX;
+import com.jgraph.io.svg.SVGGraphWriter;
+import com.jgraph.io.svg.SVGVertexRenderer;
+
+public class SVGExample extends GraphEdX {
+
+	/** the default background color, white by default */
+	protected Color background = Color.white;
+
+	/**
+	 * File chooser for exporting SVG. Note that it is lazily
+	 * instaniated, always call initFileChooser before use.
+	 */
+	protected JFileChooser SVGFileChooser = null;
+
+	public SVGExample() {
+		setJMenuBar(new SVGMenuBar(this, graphFactory));
+	}
+
+	public void setDefaultBackgroundColor(Color color) {
+		background = color;
+	}
+
+	/**
+	 * @return Returns the background.
+	 */
+	public Color getDefaultBackgroundColour() {
+		return background;
+	}
+
+	/**
+	 * Utility method that ensures the file chooser is created. Start-up time
+	 * is improved by lazily instaniating choosers.
+	 *
+	 */
+	protected void initSVGFileChooser() {
+		if (SVGFileChooser == null) {
+			SVGFileChooser = new JFileChooser();
+			FileFilter fileFilter = new FileFilter() {
+				/**
+				 * @see javax.swing.filechooser.FileFilter#accept(File)
+				 */
+				public boolean accept(File f) {
+					if (f == null)
+						return false;
+					if (f.getName() == null)
+						return false;
+					if (f.getName().endsWith(".svg"))
+						return true;
+					if (f.isDirectory())
+						return true;
+
+					return false;
+				}
+
+				/**
+				 * @see javax.swing.filechooser.FileFilter#getDescription()
+				 */
+				public String getDescription() {
+					return "SVG file (.SVG)";
+				}
+			};
+			SVGFileChooser.setFileFilter(fileFilter);
+		}
+	}
+
+	protected void exportSVG() {
+		int returnValue = JFileChooser.CANCEL_OPTION;
+		initSVGFileChooser();
+		returnValue = SVGFileChooser.showSaveDialog(graph);
+		if (returnValue == JFileChooser.APPROVE_OPTION) {
+			VertexRenderer oldRenderer = VertexView.renderer;
+			try {
+				OutputStream out = new BufferedOutputStream(new FileOutputStream(
+						SVGFileChooser.getSelectedFile()));
+				// Set all vertices to use the SVG renderer
+				VertexView.renderer = new SVGVertexRenderer();
+				SVGGraphWriter writer = new SVGGraphWriter();
+				// SVG nodes will have no size at this point. They need
+				// to be given suitable bounds prior to being passed to the
+				// layout algorithm.
+//				graph.computeVertexSizes();
+				writer.write(new BufferedOutputStream(out), null,
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
Analysis of this package determined that this line of code contains a resource that might not have closed properly. A resource leak can slow down or crash your system.

**Fix**
Consider closing the resource returned by the *BufferedOutputStream* constructor in a try-finally block. The resource is referenced at lines 104, 113, 115. The resource is closed at line 116. There are other execution paths that do not contain closure statements, for example, when OutputStream.flush throws an exception.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).